sklearn/preprocessing/imputation.py

```diff
@@ -299,13 +312,69 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
+    def _sparse_transform(self, X, missing_values, axis, valid_stats, valid_idx):
+        """transformer on sparse data."""
+        mask = _get_mask(X.data, missing_values)
+        indexes = np.repeat(np.arange(len(X.indptr) - 1, dtype=np.int),
+                            np.diff(X.indptr))[mask]
+
+        X.data[mask] = astype(valid_stats[indexes], X.dtype,
+                              copy=False)
+
+        if self.add_missing_indicator:
+            mask_matrix = X.__class__((mask, X.indices.copy(),
+                                      X.indptr.copy()), shape=X.shape,
+                                      dtype=X.dtype)
+            mask_matrix.eliminate_zeros()  # removes explicit False entries
+            features_with_missing_values = np.where(sparse_min_max
```

Comment:
Well, the getnnz approach is not available to us, but the sum seems more elegant